src/g2.rs

```diff
@@ -918,15 +921,36 @@ impl G2Projective {
     /// This is equivalent to multiplying by $h\_\textrm{eff} = 3(z^2 - 1) \cdot
     /// h_2$, where $h_2$ is the cofactor of $\mathbb{G}\_2$ and $z$ is the
     /// parameter of BLS12-381.
+    ///
+    /// The endomorphism is only actually used if the crate feature `endo` is
+    /// enabled, and it is disabled by default to mitigate potential patent
+    /// issues.
     pub fn clear_cofactor(&self) -> G2Projective {
-        let t1 = self.mul_by_x(); // [x] P
-        let t2 = self.psi(); // psi(P)
-
-        self.double().psi2() // psi^2(2P)
-            + (t1 + t2).mul_by_x() // psi^2(2P) + [x^2] P + [x] psi(P)
-            - t1 // psi^2(2P) + [x^2 - x] P + [x] psi(P)
-            - t2 // psi^2(2P) + [x^2 - x] P + [x - 1] psi(P)
-            - self // psi^2(2P) + [x^2 - x - 1] P + [x - 1] psi(P)
+        #[cfg(feature = "endo")]
+        fn clear_cofactor(this: &G2Projective) -> G2Projective {
+            let t1 = this.mul_by_x(); // [x] P
+            let t2 = this.psi(); // psi(P)
+
+            this.double().psi2() // psi^2(2P)
+                + (t1 + t2).mul_by_x() // psi^2(2P) + [x^2] P + [x] psi(P)
+                - t1 // psi^2(2P) + [x^2 - x] P + [x] psi(P)
+                - t2 // psi^2(2P) + [x^2 - x] P + [x - 1] psi(P)
+                - this // psi^2(2P) + [x^2 - x - 1] P + [x - 1] psi(P)
+        }
+
+        #[cfg(not(feature = "endo"))]
+        fn clear_cofactor(this: &G2Projective) -> G2Projective {
+            this.multiply(&[
+                0x51, 0x55, 0xa9, 0xaa, 0x5, 0x0, 0x2, 0xe8, 0xb4, 0xf6, 0xbb, 0xde, 0xa, 0x4c,
+                0x89, 0x59, 0xa3, 0xf6, 0x89, 0x66, 0xc0, 0xcb, 0x54, 0xe9, 0x1a, 0x7c, 0x47, 0xd7,
+                0x69, 0xec, 0xc0, 0x2e, 0xb0, 0x12, 0x12, 0x5d, 0x1, 0xbf, 0x82, 0x6d, 0x95, 0xdb,
+                0x31, 0x87, 0x17, 0x2f, 0x9c, 0x32, 0xe1, 0xff, 0x8, 0x15, 0x3, 0xff, 0x86, 0x99,
+                0x68, 0xd7, 0x5a, 0x14, 0xe9, 0xa8, 0xe2, 0x88, 0x28, 0x35, 0x1b, 0xa9, 0xe, 0x6a,
+                0x4c, 0x58, 0xb3, 0x75, 0xee, 0xf2, 0x8, 0x9f, 0xc6, 0xb,
```

Comment:
I have not checked this value, but `test_clear_cofactor` is unaltered and still passes.